File: cpp/bomb.cpp
#include "bomb.hpp"

std::istream& operator>>(std::istream& in, Bomb& bomb) {
  in >> bomb.owner >> bomb.position >> bomb.ticks_left;
  return in;
}

std::ostream& operator<<(std::ostream& out, const Bomb& bomb) {
  out << bomb.owner << " " << bomb.position << " " << bomb.ticks_left;
  return out;
}

File: java/Bomb.java

public class Bomb {
   public final int playerId;
   public final int ticksLeft;
   public final int x;
   public final int y;

   public Bomb(int playerId, int ticksLeft, int x, int y) {
      this.playerId = playerId;
      this.ticksLeft = ticksLeft;
      this.x = x;
      this.y = y;
   }
}
